ksqldb-common/src/main/java/io/confluent/ksql/schema/ksql/SystemColumns.java

```diff
@@ -115,11 +117,18 @@ public static boolean isPseudoColumn(
   }
 
   public static Set<ColumnName> pseudoColumnNames(final int pseudoColumnVersion) {
-    if (!PSEUDO_COLUMN_NAMES_BY_VERSION.containsKey(pseudoColumnVersion)) {
+    final Set<ColumnName> names = pseudoColumns
+        .stream()
+        .filter(col -> col.version <= pseudoColumnVersion)
+        .map(col -> col.name)
+        .collect(Collectors.toSet());
+
+    if (names.isEmpty()) {
```

Comment:
This check is no longer the same as the previous check. Previously, we were checking whether the pseudocolumn version passed was valid or not. We should continue to do that by checking that `pseudoColumnVersion` is at least `LEGACY_PSEUDOCOLUMN_VERSION_NUMBER` and no greater than `CURRENT_PSEUDOCOLUMN_VERSION_NUMBER`.